### DISCUSSION TITLE
Can't build on fresh RPi 3

### DISCUSSION THREAD

**Discussion by DanV410** [rtl-airband/RTLSDR-Airband#297] [Q&A]
I have a Rasberry Pi 3B + that I just created an SD card for tonight using the Raspberry Pi imaging tool. I am able to install rtl-fm, rtl-test, etc. and get it running, but both times I've tried to build RTLSDR-Airband on a clean install I get the following errors:

`pi@raspberrypi:~/RTLSDR-Airband-4.0.0/build $ make
Scanning dependencies of target hello_fft
[  4%] Building CXX object src/hello_fft/CMakeFiles/hello_fft.dir/mailbox.c.o
[  9%] Building CXX object src/hello_fft/CMakeFiles/hello_fft.dir/gpu_fft.c.o
[ 13%] Building CXX object src/hello_fft/CMakeFiles/hello_fft.dir/gpu_fft_twiddles.c.o
/home/pi/RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft_twiddles.c: In function ‘float* twiddles_step_16(double, float*, double)’:
/home/pi/RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft_twiddles.c:70:39: warning: unused parameter ‘two_pi’ [-Wunused-parameter]
   70 | static float *twiddles_step_16(double two_pi, float *out, double theta)                                                        {
      |                                ~~~~~~~^~~~~~
[ 18%] Building CXX object src/hello_fft/CMakeFiles/hello_fft.dir/gpu_fft_shaders.c.o
[ 22%] Building CXX object src/hello_fft/CMakeFiles/hello_fft.dir/gpu_fft_base.c.o
/home/pi/RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft_base.c: In function ‘unsigned int gpu_fft_base_exec_direct(GPU_FFT_BASE*, int)’:
/home/pi/RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft_base.c:65:16: warning: comparison of integer expressions of different signedness: ‘unsigned int’ and ‘int’ [-Wsign-compare]
   65 |     for (q=0; q<num_qpus; q++) { // Launch shader(s)
      |               ~^~~~~~~~~
/home/pi/RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft_base.c:72:50: warning: comparison of integer expressions of different signedness: ‘unsigned int’ and ‘int’ [                                                       -Wsign-compare]
   72 |         if (((base->peri[V3D_SRQCS]>>16) & 0xff) == num_qpus) break; //                                                        All done?
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/home/pi/RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft_base.c:54:17: warning: unused variable ‘t’ [-Wunused-variable]
   54 |     unsigned q, t;
      |                 ^
[ 27%] Linking CXX static library libhello_fft.a
[ 27%] Built target hello_fft
[ 31%] Generating version.cpp, _version.cpp
Scanning dependencies of target rtl_airband_base
[ 36%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/config.cpp.o
[ 40%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/input-common.cpp.                                                       o
[ 45%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/input-file.cpp.o
[ 50%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/input-helpers.cpp                                                       .o
[ 54%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/mixer.cpp.o
[ 59%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/output.cpp.o
[ 63%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/rtl_airband.cpp.o
[ 68%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/squelch.cpp.o
[ 72%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/util.cpp.o
[ 77%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/udp_stream.cpp.o
[ 77%] Generating version.cpp, _version.cpp
[ 81%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/version.cpp.o
[ 86%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/input-rtlsdr.cpp.                                                       o
[ 90%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/pulse.cpp.o
[ 95%] Building ASM object src/CMakeFiles/rtl_airband_base.dir/rtl_airband_neon.                                                       s.o
[ 95%] Built target rtl_airband_base
Scanning dependencies of target rtl_airband
[100%] Linking CXX executable rtl_airband
/usr/bin/ld: hello_fft/libhello_fft.a(gpu_fft_base.c.o): in function `gpu_fft_al                                                       loc(int, unsigned int, GPU_FFT_PTR*)':
gpu_fft_base.c:(.text+0x278): undefined reference to `bcm_host_get_sdram_address                                                       '
/usr/bin/ld: gpu_fft_base.c:(.text+0x2a0): undefined reference to `bcm_host_get_                                                       peripheral_address'
/usr/bin/ld: gpu_fft_base.c:(.text+0x2a8): undefined reference to `bcm_host_get_                                                       peripheral_size'
/usr/bin/ld: hello_fft/libhello_fft.a(gpu_fft_base.c.o): in function `gpu_fft_ba                                                       se_release(GPU_FFT_BASE*)':
gpu_fft_base.c:(.text+0x33c): undefined reference to `bcm_host_get_peripheral_si                                                       ze'
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/rtl_airband.dir/build.make:120: src/rtl_airband] Er                                                       ror 1
make[1]: *** [CMakeFiles/Makefile2:136: src/CMakeFiles/rtl_airband.dir/all] Erro                                                       r 2
make: *** [Makefile:149: all] Error 2`

Here is my configuration:

pi@raspberrypi:~/RTLSDR-Airband-4.0.0/build $ cmake -DPLATFORM=rpiv2 -DSOAPYSDR=                                                       OFF -DNFM=ON ../
-- The CXX compiler identification is GNU 10.2.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build type not specified: defaulting to Release
-- Performing Test CXX_HAS_PTHREAD
-- Performing Test CXX_HAS_PTHREAD - Success
-- Performing Test CXX_HAS_FFAST_MATH
-- Performing Test CXX_HAS_FFAST_MATH - Success
-- Looking for sincosf
-- Looking for sincosf - found
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2")
-- Checking for module 'libconfig++'
--   Found libconfig++, version 1.5
-- Found lame includes: /usr/include/lame/lame.h
-- Found lame library: /usr/lib/arm-linux-gnueabihf/libmp3lame.so
-- Checking for module 'shout'
--   Found shout, version 2.4.5
-- Looking for SHOUT_TLS_AUTO
-- Looking for SHOUT_TLS_AUTO - found
-- Looking for SHOUT_TLS_AUTO_NO_PLAIN
-- Looking for SHOUT_TLS_AUTO_NO_PLAIN - found
-- Looking for SHOUT_TLS_RFC2818
-- Looking for SHOUT_TLS_RFC2818 - found
-- Looking for SHOUT_TLS_RFC2817
-- Looking for SHOUT_TLS_RFC2817 - found
-- Looking for SHOUT_TLS_DISABLED
-- Looking for SHOUT_TLS_DISABLED - found
-- Looking for shout_set_tls
-- Looking for shout_set_tls - found
-- Checking for module 'librtlsdr'
--   Found librtlsdr, version 0.6.0-32-gd770
-- Found RTLSDR: /usr/local/lib/librtlsdr.so
-- Checking for module 'libmirisdr'
--   No package 'libmirisdr' found
-- Could NOT find MiriSDR (missing: MIRISDR_LIBRARY MIRISDR_INCLUDE_DIR)
-- Checking for module 'libpulse'
--   Found libpulse, version 14.2
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/c++
-- Found BCM_VC: /usr/lib/arm-linux-gnueabihf/libbcm_host.so
-- RTLSDR-Airband configuration summary:

-- - Build type:                Release
-- - Operating system:          Linux
-- - SDR drivers:
--   - librtlsdr:               requested: ON, enabled: TRUE
--   - mirisdr:                 requested: ON, enabled: FALSE
--   - soapysdr:                requested: OFF, enabled: FALSE
-- - Other options:
--   - Platform:                rpiv2
--   - Broadcom VideoCore GPU:  TRUE
--   - NFM support:             ON
--   - PulseAudio:              requested: ON, enabled: TRUE
--   - Profiling:               requested: OFF, enabled: FALSE
--   - Icecast TLS support:     TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /home/pi/RTLSDR-Airband-4.0.0/build

And some details of my install:

pi@raspberrypi:~/RTLSDR-Airband-4.0.0/build $ uname -a
Linux raspberrypi 5.10.63-v7+ #1459 SMP Wed Oct 6 16:41:10 BST 2021 armv7l GNU/Linux

pi@raspberrypi:~/RTLSDR-Airband-4.0.0/build $ cat /proc/device-tree/model
Raspberry Pi 3 Model B Plus Rev 1.3pi@raspberrypi:~/RTLSDR-Airband-4.0.0/build $

pi@raspberrypi:~/RTLSDR-Airband-4.0.0/build $ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 11 (bullseye)"
NAME="Raspbian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

I've installed all of the dependancies as described on the wiki, with no errors. Any ideas?

**Comment by charlie-foxtrot** [comment]
I haven't updated to bullseye yet, but here are the `cmake` flags that work for me on a rpi4:

```
-DPLATFORM=native -DNFM=TRUE -DRTLSDR=TRUE -DMIRISDR=FALSE -DSOAPYSDR=FALSE -DPULSEAUDIO=FALSE
```

**Comment by DanV410** [comment]
Definitely a bullseye thing. Same hardware, same commands work just fine on
the older
Raspbian I downloaded just now.

Dan

On Thu, Nov 11, 2021 at 10:01 PM charlie-foxtrot ***@***.***>
wrote:

> I haven't updated to bullseye yet, but here are the cmake flags that work
> for me on a rpi4:
>
> -DPLATFORM=native -DNFM=TRUE -DRTLSDR=TRUE -DMIRISDR=FALSE -DSOAPYSDR=FALSE -DPULSEAUDIO=FALSE
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/szpajder/RTLSDR-Airband/discussions/297#discussioncomment-1629116>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFKY5XDZPL2DA2SSOVQLP33ULR7PNANCNFSM5H32UDTA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
Dan Verbus
***@***.***

**Comment by szpajder** [comment]
Looks like the GPU_FFT code needs an update. I'll look into this when I'll have a spare moment. Meanwhile just build with `PLATFORM=armv7-generic`.

**Comment by szpajder** [comment]
Fix pushed to unstable branch

**Comment by DanV410** [comment]
That was fast, thank you!

On Sun, Nov 14, 2021 at 5:14 AM Tomasz Lemiech ***@***.***>
wrote:

> Fix pushed to unstable branch
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/szpajder/RTLSDR-Airband/discussions/297#discussioncomment-1639444>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFKY5XBOY54BADZRUL5FV73UL6DXZANCNFSM5H32UDTA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Dan Verbus
***@***.***

**Comment by szpajder** [comment]
Fix released in v4.0.1.
